doc/user/tutorials/making_your_first_app.rst

```diff
@@ -0,0 +1,510 @@
+simply put it in the :file:`myapp/plugins.py` file.
+
+.. note:: For "serious" plugins, we recommend mirroring the ``armi/physics/[subphysics]``
+    structure of the ARMI framework physics plugin folder.
+
+We will start the plugin simply by pointing to the two physics kernels we wish to
+register. We hook them in and tell ARMI the ``ORDER`` they should be run in based on
+the built-in ``STACK_ORDER`` attribute. We will come back to this plugin definition later
+on to do add a little more to the plugin. 
+
+
+.. code-block:: python
+    :caption: ``myapp/plugin.py``
+
+    from armi import plugins
+    from armi import interfaces
+    from armi.interfaces import STACK_ORDER as ORDER
+
+    from myapp import fluxSolver
+    from myapp import thermalSolver
+
+
+    class DummyPhysicsPlugin(plugins.ArmiPlugin):
+        @staticmethod
+        @plugins.HOOKIMPL
+        def exposeInterfaces(cs):
+            kernels = [
+                interfaces.InterfaceInfo(ORDER.FLUX, fluxSolver.FluxInterface, {}),
+                interfaces.InterfaceInfo(ORDER.THERMAL_HYDRAULICS, thermalSolver.ThermalInterface, {}),
+            ]
+            return kernels
+
+Creating the physics kernels
+============================
+So far we have basically been weaving an administrative thread to tell ARMI about the code
+we want to run. Now we finally get to write the guts of the code that actually does
+something. In your real app, this code will run your own industrial or research code, or
+perform your own methodology.  Here we just have it make up dummy values representing flux
+and temperatures.
+
+Making the dummy flux kernel
+----------------------------
+In a previous tutorial, we made a function that sets a dummy flux to all parts of the core
+based on a radial distance from the origin. Here we will re-use that code but package it
+more formally so that ARMI can actually run it for us from a user perspective.
+
+The interface is responsible largely for scheduling activities to run at various time
+points. For a flux calculation, we want it to compute at every single time node, so we use
+the :py:meth:`armi.interfaces.Interface.interactEveryNode` hook.
+
+These interaction hooks can call arbitrarily complex code. The code could, for example:
+
+* Run an external executable locally
+* Submit an external code to a cloud HPC and wait for it to complete
+* Run an internal physics tool
+
+Here it just a tiny bit of math locally.
+
+.. code-block:: python
+    :caption: ``myapp/fluxSolver.py``
+
+    import os
+
+    import numpy as np
+
+    from armi import runLog
+    from armi import interfaces
+    from armi.physics import neutronics
+
+
+    class FluxInterface(interfaces.Interface):
+        name = "dummyFlux"
+
+        def interactEveryNode(self, cycle=None, timeNode=None):
+            runLog.info("Computing neutron flux and power.")
+            setFakePower(self.r.core)
+
+
+    def setFakePower(core):
```

Comment:
Add some docstrings to set a good precedence for documentation of functions.

Reply:
I considered that but am trying to keep it super concise.